### DISCUSSION TITLE
how to close agent enhance for redisson?

### DISCUSSION THREAD

**Discussion by treebreath** [apache/skywalking#12237] [Q&A] [answered]
Is it possible close agent enhance for redisson? how to configure for it. Your help would be appreciated.

**Comment by wu-sheng** [comment, marked as answer]
We can disable relative plugins.

- Use plugin.exclude_plugins https://skywalking.apache.org/docs/skywalking-java/latest/en/setup/service-agent/java-agent/configurations/
- Remove plugin jar file from the folder. https://skywalking.apache.org/docs/skywalking-java/latest/en/setup/service-agent/java-agent/readme/#plugins

**Reply by treebreath** [reply]
Thanks for your reply. Let me ask another question. I found the following warning in the agent log: Will this cause problems? 
`WARN 2024-05-18 12:36:26.568 main AbstractClassEnhancePluginDefine : enhance class com.mysql.cj.jdbc.ServerPreparedStatement by plugin org.apache.skywalking.apm.plugin.jdbc.mysql.v6.define.PreparedStatementNullSetterInstrumentation is not activated. Witness class com.mysql.cj.api.MysqlConnection does not exist. 
WARN 2024-05-18 12:36:33.705 main AbstractClassEnhancePluginDefine : enhance class org.springframework.web.method.support.InvocableHandlerMethod by plugin org.apache.skywalking.apm.plugin.spring.mvc.v3.define.InvocableHandlerInstrumentation is not activated. Witness class org.springframework.web.servlet.view.xslt.AbstractXsltView does not exist. `
Does this mean that the MYSQL proxy enhancement failed and spring controller method proxy enhancement failed?

**Reply by wu-sheng** [reply]
This isn't an issue. This is message meaning, plugins exist but no need to work. Mostly because of versions. 

When you understand SkyWalking more, you will be able to tell which plugins you need, rather than putting all things by default.

**Reply by treebreath** [reply]
When I start my java program, the error log occurs in skywalking-api.log.  Is it matter?  I noticed that another issue also mentioned a similar problem https://github.com/apache/skywalking/issues/7040, but that problem was because the authentication function was turned on, which I did not do here.
`grpc-default-executor-18 LogReportServiceClient : Try to send 4 log data to collect
or, with unexpected exception. 
org.apache.skywalking.apm.dependencies.io.grpc.StatusRuntimeException: UNAVAILABLE: Channel shutdown invoked
        at org.apache.skywalking.apm.dependencies.io.grpc.Status.asRuntimeException(Status.java:539)
        at org.apache.skywalking.apm.dependencies.io.grpc.stub.ClientCalls$StreamObserverToCallListenerAdapter.on
Close(ClientCalls.java:487)
        at org.apache.skywalking.apm.dependencies.io.grpc.internal.ClientCallImpl.closeObserver(ClientCallImpl.ja
va:576)
        at org.apache.skywalking.apm.dependencies.io.grpc.internal.ClientCallImpl.access$300(ClientCallImpl.java:
70)
        at org.apache.skywalking.apm.dependencies.io.grpc.internal.ClientCallImpl$ClientStreamListenerImpl$1Strea
mClosed.runInternal(ClientCallImpl.java:757)
        at org.apache.skywalking.apm.dependencies.io.grpc.internal.ClientCallImpl$ClientStreamListenerImpl$1Strea
mClosed.runInContext(ClientCallImpl.java:736)
        at org.apache.skywalking.apm.dependencies.io.grpc.internal.ContextRunnable.run(ContextRunnable.java:37)
        at org.apache.skywalking.apm.dependencies.io.grpc.internal.SerializingExecutor.run(SerializingExecutor.ja
va:133)
        at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
        at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
        at java.base/java.lang.Thread.run(Thread.java:840)`

**Reply by wu-sheng** [reply]
> UNAVAILABLE: Channel shutdown invoked at 

This only means connection is closed by server side. It could be caused by authentication or firewall, even unstable network.
